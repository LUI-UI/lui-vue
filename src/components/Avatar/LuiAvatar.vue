<script setup lang="ts">
import type { Filter, Color, Size, Border, Text, NarrowedVariant, Rounded } from '@/globals/types'

import type {
  TwClassInterface,
  LayoutInterface,
  FlexGridInterface,
  SizingInterface,
  BordersInterface
} from '@/globals/interfaces'

import { computed, toRefs } from 'vue'
import type { PropType } from 'vue'

import { useGlobalColorClasses } from '../../composables'

const props = defineProps({
  variant: {
    type: String as PropType<NarrowedVariant>,
    default: 'solid'
  },
  color: {
    type: String as PropType<Color>,
    default: 'primary'
  },
  filter: {
    type: String as PropType<Filter>,
    default: 'none'
  },
  size: {
    type: String as PropType<Size>,
    default: 'md'
  },
  border: {
    type: Boolean as PropType<Border>,
    default: false
  },
  rounded: {
    type: [Boolean, String] as PropType<Rounded>,
    default: false
  },
  text: {
    type: String as PropType<Text>,
    default: ''
  },
  src: {
    type: String,
    default: ''
  },
  alt: {
    type: String,
    default: ''
  }
})

const { backgroundColorClasses, textColorClasses, borderColorClasses } = useGlobalColorClasses(
  toRefs(props)
)

const computedImageClasses = computed(() => {
  const containerClasses: LayoutInterface | SizingInterface | BordersInterface = {
    borderWidth: 'border',
    borderStyle: 'border-solid',
    display: 'block',
    width: 'w-full',
    height: 'h-full',
    objectFit: 'object-contain',
    borderRadius: {
      'rounded-lg': props.rounded === true,
      'rounded-full': props.rounded === 'full'
    },
    borderColor: borderColorClasses.value
  }
  return Object.values(containerClasses)
})

const computedIconClasses = computed(() => {
  const iconClasses: LayoutInterface | FlexGridInterface = {
    display: 'flex',
    justifyContent: 'justify-center',
    alignItems: 'items-center'
  }
  return Object.values(iconClasses)
})

const computedAvatarClasses = computed(() => {
  const avatarClasses: TwClassInterface = {
    display: 'flex',
    borderRadius: {
      'rounded-lg': props.rounded === true,
      'rounded-full': props.rounded === 'full'
    },
    borderWidth: 'border',
    borderStyle: 'border-solid',
    justifyContent: 'justify-center',
    alignItems: 'items-center',
    // padding:
    //   props.text.length > 0 || !!slots.icon
    //     ? {
    //         "px-0.5": props.size === "xs",
    //         "px-1": props.size === "sm",
    //         "px-1.5": props.size === "md",
    //         "px-2": props.size === "lg",
    //         "px-2.5": props.size === "xl",
    //       }
    //     : null,
    width: {
      'w-6': props.size === 'xs',
      'w-8': props.size === 'sm',
      'w-12': props.size === 'md',
      'w-16': props.size === 'lg',
      'w-20': props.size === 'xl'
    },

    height: {
      'h-6': props.size === 'xs',
      'h-8': props.size === 'sm',
      'h-12': props.size === 'md',
      'h-16': props.size === 'lg',
      'h-20': props.size === 'xl'
    },
    fontSize: {
      'text-sm': props.size === 'xs' || props.size === 'sm',
      'text-xl': props.size === 'md',
      'text-3xl': props.size === 'lg' || props.size === 'xl'
    },
    backgroundColor: props.src.length === 0 ? backgroundColorClasses.value : null,
    textColor: textColorClasses.value,
    borderColor: props.src.length === 0 ? borderColorClasses.value : 'border-transparent'
  }
  return Object.values(avatarClasses)
})
</script>
<template>
  <div class="lui-avatar" :class="computedAvatarClasses">
    <img :src="src" :alt="alt" :class="computedImageClasses" v-if="src.length > 0" />
    <span :class="computedIconClasses" v-if="$slots.icon"><slot name="icon"></slot></span>
    <span v-if="text.length > 0" ref="avatarContent">{{ text }}</span>
  </div>
</template>

<script lang="ts">
export default {
  name: 'LuiAvatar'
}
</script>
