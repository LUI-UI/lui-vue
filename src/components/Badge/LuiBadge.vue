<!-- <script setup lang="ts">
import type {
  Filter,
  Color,
  Size,
  Border,
  Text,
  NarrowedVariant,
} from "@/globals/types";
import { ref, computed, toRefs, useSlots } from "vue";
import type { PropType } from "vue";
import type {
  TwClassInterface,
  LayoutInterface,
  FlexGridInterface,
} from "@/globals/interfaces";
import { useGlobalColorClasses } from "../../composables";
type Position = "top-left" | "top-right" | "bottom-left" | "bottom-right";

const slots = useSlots();
const props = defineProps({
  variant: {
    type: String as PropType<NarrowedVariant>,
    default: "solid",
  },
  color: {
    type: String as PropType<Color>,
    default: "primary",
  },
  filter: {
    type: String as PropType<Filter>,
    default: "none",
  },
  size: {
    type: String as PropType<Size>,
    default: "md",
  },
  border: {
    type: Boolean as PropType<Border>,
    default: false,
  },
  text: {
    type: String as PropType<Text>,
    default: "",
  },
  position: {
    type: String as PropType<Position>,
    default: "bottom-right",
  },
});
const { backgroundColorClasses, textColorClasses, borderColorClasses } =
  useGlobalColorClasses(toRefs(props));
const badgeWrapper = ref<HTMLDivElement>();
const badgeContent = ref<HTMLSpanElement>();

// const overflow = ref(false);
const overflow = computed(() => props.text.length > 2);

// watchEffect(() => {
//   const wrapperWidh = badgeWrapper.value?.scrollWidth;
//   const contentWidth = badgeContent.value?.scrollWidth;
//   if (wrapperWidh && contentWidth) {
//     if (contentWidth > wrapperWidh) {
//       overflow.value = true;
//     } else {
//       overflow.value = false;
//     }
//   }
// });
const computedContainerClasses = computed(() => {
  const containerClasses: LayoutInterface = {
    position: "relative",
    display: "inline-block",
  };
  return Object.values(containerClasses);
});
const computedIconClasses = computed(() => {
  const iconClasses: LayoutInterface | FlexGridInterface = {
    display: "flex",
    justifyContent: "justify-center",
    alignItems: "items-center",
  };
  return Object.values(iconClasses);
});
const computedBadgeClasses = computed(() => {
  const badgeClasses: TwClassInterface = {
    display: overflow.value ? "inline-block" : "flex",
    position: { absolute: !!slots.default },
    top: slots.default
      ? {
          "top-0":
            props.position === "top-left" || props.position === "top-right",
        }
      : null,
    bottom: slots.default
      ? {
          "bottom-0":
            props.position === "bottom-left" ||
            props.position === "bottom-right",
        }
      : null,
    left: slots.default
      ? {
          "left-0":
            props.position === "bottom-left" || props.position === "top-left",
        }
      : null,
    right: slots.default
      ? {
          "right-0":
            props.position === "bottom-right" || props.position === "top-right",
        }
      : null,
    translate: slots.default
      ? {
          "-translate-y-1/2 -translate-x-1/2": props.position === "top-left",
          "-translate-y-1/2 translate-x-1/2": props.position === "top-right",
          "translate-y-1/2 -translate-x-1/2": props.position === "bottom-left",
          "translate-y-1/2 translate-x-1/2": props.position === "bottom-right",
        }
      : null,
    borderRadius: "rounded-full",
    borderWidth: "border",
    borderStyle: "border-solid",
    justifyContent: { ["justify-center"]: !overflow.value },
    alignItems: { ["items-center"]: !overflow.value },
    padding:
      props.text.length > 0 || !!slots.icon
        ? {
            "px-0.5": props.size === "xs",
            "px-1": props.size === "sm",
            "px-1.5": props.size === "md",
            "px-2": props.size === "lg",
            "px-2.5": props.size === "xl",
          }
        : null,
    width:
      props.text.length > 0 || !!slots.icon
        ? !overflow.value
          ? {
              "w-3": props.size === "xs",
              "w-4": props.size === "sm",
              "w-5": props.size === "md",
              "w-6": props.size === "lg",
              "w-7": props.size === "xl",
            }
          : null
        : {
            "w-1.5": props.size === "xs",
            "w-2": props.size === "sm",
            "w-2.5": props.size === "md",
            "w-3": props.size === "lg",
            "w-3.5": props.size === "xl",
          },

    height:
      props.text.length > 0 || !!slots.icon
        ? !overflow.value
          ? {
              "h-3": props.size === "xs",
              "h-4": props.size === "sm",
              "h-5": props.size === "md",
              "h-6": props.size === "lg",
              "h-7": props.size === "xl",
            }
          : null
        : {
            "h-1.5": props.size === "xs",
            "h-2": props.size === "sm",
            "h-2.5": props.size === "md",
            "h-3": props.size === "lg",
            "h-3.5": props.size === "xl",
          },
    fontSize: {
      "text-xs": props.size === "xs" || props.size === "sm",
      "text-sm": props.size === "md",
      "text-base": props.size === "lg" || props.size === "xl",
    },
    backgroundColor: backgroundColorClasses.value,
    textColor: textColorClasses.value,
    borderColor: borderColorClasses.value,
  };
  return Object.values(badgeClasses);
});
</script> -->
<template>
  <div class="lui-badge" :class="computedContainerClasses">
    <slot></slot>
    <div ref="badgeWrapper" :class="computedBadgeClasses">
      <span :class="computedIconClasses" v-if="$slots.icon"><slot name="icon"></slot></span>
      <span v-if="text.length > 0" ref="badgeContent">{{ text }}</span>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import type { Filter, Color, Size, Border, Text, NarrowedVariant } from '@/globals/types'
import { ref, computed, toRefs } from 'vue'
import type { PropType } from 'vue'
import type { TwClassInterface, LayoutInterface, FlexGridInterface } from '@/globals/interfaces'
import { useGlobalColorClasses } from '../../composables'
type Position = 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right'

export default defineComponent({
  name: 'LuiBadge',
  props: {
    variant: {
      type: String as PropType<NarrowedVariant>,
      default: 'solid'
    },
    color: {
      type: String as PropType<Color>,
      default: 'primary'
    },
    filter: {
      type: String as PropType<Filter>,
      default: 'none'
    },
    size: {
      type: String as PropType<Size>,
      default: 'md'
    },
    border: {
      type: Boolean as PropType<Border>,
      default: false
    },
    text: {
      type: String as PropType<Text>,
      default: ''
    },
    position: {
      type: String as PropType<Position>,
      default: 'bottom-right'
    }
  },
  setup(props, { slots }) {
    const { backgroundColorClasses, textColorClasses, borderColorClasses } = useGlobalColorClasses(
      toRefs(props)
    )
    const badgeWrapper = ref<HTMLDivElement>()
    const badgeContent = ref<HTMLSpanElement>()

    // const overflow = ref(false);
    const overflow = computed(() => props.text.length > 2)

    // watchEffect(() => {
    //   const wrapperWidh = badgeWrapper.value?.scrollWidth;
    //   const contentWidth = badgeContent.value?.scrollWidth;
    //   if (wrapperWidh && contentWidth) {
    //     if (contentWidth > wrapperWidh) {
    //       overflow.value = true;
    //     } else {
    //       overflow.value = false;
    //     }
    //   }
    // });
    const computedContainerClasses = computed(() => {
      const containerClasses: LayoutInterface = {
        position: 'relative',
        display: 'inline-block'
      }
      return Object.values(containerClasses)
    })
    const computedIconClasses = computed(() => {
      const iconClasses: LayoutInterface | FlexGridInterface = {
        display: 'flex',
        justifyContent: 'justify-center',
        alignItems: 'items-center'
      }
      return Object.values(iconClasses)
    })
    const computedBadgeClasses = computed(() => {
      const badgeClasses: TwClassInterface = {
        display: overflow.value ? 'inline-block' : 'flex',
        position: { absolute: !!slots.default },
        top: slots.default
          ? {
              'top-0': props.position === 'top-left' || props.position === 'top-right'
            }
          : null,
        bottom: slots.default
          ? {
              'bottom-0': props.position === 'bottom-left' || props.position === 'bottom-right'
            }
          : null,
        left: slots.default
          ? {
              'left-0': props.position === 'bottom-left' || props.position === 'top-left'
            }
          : null,
        right: slots.default
          ? {
              'right-0': props.position === 'bottom-right' || props.position === 'top-right'
            }
          : null,
        translate: slots.default
          ? {
              '-translate-y-1/2 -translate-x-1/2': props.position === 'top-left',
              '-translate-y-1/2 translate-x-1/2': props.position === 'top-right',
              'translate-y-1/2 -translate-x-1/2': props.position === 'bottom-left',
              'translate-y-1/2 translate-x-1/2': props.position === 'bottom-right'
            }
          : null,
        borderRadius: 'rounded-full',
        borderWidth: 'border',
        borderStyle: 'border-solid',
        justifyContent: { ['justify-center']: !overflow.value },
        alignItems: { ['items-center']: !overflow.value },
        padding:
          props.text.length > 0 || !!slots.icon
            ? {
                'px-0.5': props.size === 'xs',
                'px-1': props.size === 'sm',
                'px-1.5': props.size === 'md',
                'px-2': props.size === 'lg',
                'px-2.5': props.size === 'xl'
              }
            : null,
        width:
          props.text.length > 0 || !!slots.icon
            ? !overflow.value
              ? {
                  'w-3': props.size === 'xs',
                  'w-4': props.size === 'sm',
                  'w-5': props.size === 'md',
                  'w-6': props.size === 'lg',
                  'w-7': props.size === 'xl'
                }
              : null
            : {
                'w-1.5': props.size === 'xs',
                'w-2': props.size === 'sm',
                'w-2.5': props.size === 'md',
                'w-3': props.size === 'lg',
                'w-3.5': props.size === 'xl'
              },

        height:
          props.text.length > 0 || !!slots.icon
            ? !overflow.value
              ? {
                  'h-3': props.size === 'xs',
                  'h-4': props.size === 'sm',
                  'h-5': props.size === 'md',
                  'h-6': props.size === 'lg',
                  'h-7': props.size === 'xl'
                }
              : null
            : {
                'h-1.5': props.size === 'xs',
                'h-2': props.size === 'sm',
                'h-2.5': props.size === 'md',
                'h-3': props.size === 'lg',
                'h-3.5': props.size === 'xl'
              },
        fontSize: {
          'text-xs': props.size === 'xs' || props.size === 'sm',
          'text-sm': props.size === 'md',
          'text-base': props.size === 'lg' || props.size === 'xl'
        },
        backgroundColor: backgroundColorClasses.value,
        textColor: textColorClasses.value,
        borderColor: borderColorClasses.value
      }
      return Object.values(badgeClasses)
    })
    return {
      computedBadgeClasses,
      computedIconClasses,
      computedContainerClasses,
      badgeWrapper,
      badgeContent,
      backgroundColorClasses,
      textColorClasses,
      borderColorClasses
    }
  }
})
</script>
