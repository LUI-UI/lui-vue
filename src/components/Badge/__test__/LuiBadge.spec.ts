import { afterEach, beforeEach, describe, expect, it } from 'vitest'
import type { VueWrapper } from '@vue/test-utils'
import { mount } from '@vue/test-utils'
import LuiBadge from '../LuiBadge.vue'
import LuiAvatar from '../../Avatar/LuiAvatar.vue'
import type { Color, Filter, Size } from '@/globals/types'

interface filteredBadge {
  color: Color
  filter: Filter
  expectedClasses: {
    bg: string
    text: string
    border: string
  }
}
const filteredBadges: filteredBadge[] = [
  {
    color: 'primary',
    filter: 'darken',
    expectedClasses: {
      bg: 'bg-primary-800',
      text: 'text-primary-100',
      border: 'border-primary-100',
    },
  },
  {
    color: 'secondary',
    filter: 'darken',
    expectedClasses: {
      bg: 'bg-secondary-800',
      text: 'text-secondary-100',
      border: 'border-secondary-100',
    },
  },
  {
    color: 'info',
    filter: 'darken',
    expectedClasses: {
      bg: 'bg-info-800',
      text: 'text-info-100',
      border: 'border-info-100',
    },
  },
  {
    color: 'success',
    filter: 'darken',
    expectedClasses: {
      bg: 'bg-success-800',
      text: 'text-success-100',
      border: 'border-success-100',
    },
  },
  {
    color: 'warning',
    filter: 'darken',
    expectedClasses: {
      bg: 'bg-warning-800',
      text: 'text-warning-100',
      border: 'border-warning-100',
    },
  },
  {
    color: 'danger',
    filter: 'darken',
    expectedClasses: {
      bg: 'bg-danger-800',
      text: 'text-danger-100',
      border: 'border-danger-100',
    },
  },
  {
    color: 'primary',
    filter: 'lighten',
    expectedClasses: {
      bg: 'bg-primary-100',
      text: 'text-primary-500',
      border: 'border-primary-500',
    },
  },
  {
    color: 'secondary',
    filter: 'lighten',
    expectedClasses: {
      bg: 'bg-secondary-100',
      text: 'text-secondary-500',
      border: 'border-secondary-500',
    },
  },
  {
    color: 'info',
    filter: 'lighten',
    expectedClasses: {
      bg: 'bg-info-100',
      text: 'text-info-500',
      border: 'border-info-500',
    },
  },
  {
    color: 'success',
    filter: 'lighten',
    expectedClasses: {
      bg: 'bg-success-100',
      text: 'text-success-500',
      border: 'border-success-500',
    },
  },
  {
    color: 'warning',
    filter: 'lighten',
    expectedClasses: {
      bg: 'bg-warning-100',
      text: 'text-warning-500',
      border: 'border-warning-500',
    },
  },
  {
    color: 'danger',
    filter: 'lighten',
    expectedClasses: {
      bg: 'bg-danger-100',
      text: 'text-danger-500',
      border: 'border-danger-500',
    },
  },
]
export function getFilteredColor(filter: string): string {
  const filterMap: Record<string, string> = {
    none: '500',
    lighten: '100',
    darken: '800',
  }

  // Return the corresponding color suffix, defaulting to "500" for unknown filters
  return filterMap[filter] || '500'
}
const colors: Color[] = [
  'primary',
  'secondary',
  'info',
  'success',
  'warning',
  'danger',
]

const expectedDefaultProps = {
  variant: 'solid',
  color: 'primary',
  filter: 'none',
  size: 'md',
  border: false,
  text: '',
  position: 'bottom-right',
}

describe('Default LuiBadge', () => {
  let wrapper: VueWrapper<any>
  beforeEach(() => {
    wrapper = mount(LuiBadge as any)
  })

  afterEach(() => {
    // Clean up after each test
    wrapper.unmount()
  })
  wrapper = mount(LuiBadge)
  it('renders properly based on default values', () => {
    // Check if the component exists
    expect(wrapper.exists()).toBe(true)

    // Check if the rendered HTML matches the expected HTML
    expect(wrapper.html()).toMatchSnapshot()

    // Check if the default props given correctly
    expect(wrapper.props()).toEqual(expectedDefaultProps)
  })
})
describe('LuiBadge class tests', () => {
  let wrapper: VueWrapper<any>
  beforeEach(() => {
    wrapper = mount(LuiBadge)
  })

  afterEach(() => {
    // Clean up after each test
    wrapper.unmount()
  })
  it('renders solid variant properly', async () => {
    await wrapper.setProps({ variant: 'solid' })
    // Check if background class rendering properly according to the variant
    expect(wrapper.find('[data-testid="lui-badge-wrapper"]').classes()).toContain(
      `bg-${wrapper.props().color}-${getFilteredColor(wrapper.props().filter)}`,
    )
  })
  it('renders outline variant properly', async () => {
    await wrapper.setProps({ variant: 'outline' })
    // Check if background class rendering properly according to the variant
    expect(wrapper.find('[data-testid="lui-badge-wrapper"]').classes()).not.toContain(
      'bg',
    )
    // Check if border class rendering properly according to the variant
    expect(wrapper.find('[data-testid="lui-badge-wrapper"]').classes()).toContain(
      `border-${wrapper.props().color}-${getFilteredColor(
        wrapper.props().filter,
      )}`,
    )
  })
  it('should render sizes correctly', async () => {
    const sizesAndValues = {
      xs: 1.5,
      sm: 2,
      md: 2.5,
      lg: 3,
      xl: 3.5,
    }

    for (const [key, value] of Object.entries(sizesAndValues)) {
      // Use `await` when setting props
      await wrapper.setProps({
        size: key as Size,
      })

      expect(wrapper.find('[data-testid="lui-badge-wrapper"]').classes()).toContain(
        `w-${value}`,
      )
      expect(wrapper.find('[data-testid="lui-badge-wrapper"]').classes()).toContain(
        `h-${value}`,
      )
    }
  })
  it('renders properly based on the color attribute', async () => {
    for (const colorOption of colors) {
      await wrapper.setProps({
        color: colorOption,
      })

      expect(wrapper.find('[data-testid="lui-badge-wrapper"]').classes()).toContain(
        `bg-${colorOption}-500`,
      )
    }
  })
  it('applies filters properly', async () => {
    for (const badge of filteredBadges) {
      await wrapper.setProps({
        color: badge.color,
        filter: badge.filter,
      })
      expect(wrapper.find('[data-testid="lui-badge-wrapper"]').classes()).contain(
        badge.expectedClasses.bg,
      )
      expect(wrapper.find('[data-testid="lui-badge-wrapper"]').classes()).contain(
        badge.expectedClasses.text,
      )
    }
  })

  it('displays filtered avatars\' border properly ', async () => {
    // Check the border of filtered avatars
    for (const badge of filteredBadges) {
      await wrapper.setProps({
        color: badge.color,
        filter: badge.filter,
        border: true,
      })
      expect(wrapper.find('[data-testid="lui-badge-wrapper"]').classes()).contain(
        badge.expectedClasses.border,
      )
    }
    // Check the border of unfiltered avatars
    for (const colorOption of colors) {
      await wrapper.setProps({
        color: colorOption,
        border: true,
      })
      expect(wrapper.find('[data-testid="lui-badge-wrapper"]').classes()).contain(
        'border',
      )
      expect(wrapper.find('[data-testid="lui-badge-wrapper"]').classes()).contain(
        'border-solid',
      )
    }
  })
})

it('displays text properly', () => {
  const testText = 'T'
  const wrapper = mount(LuiBadge, {
    props: {
      text: testText,
    },
  })
  expect(wrapper.find('[data-testid=\'lui-badge-text\']').exists()).toBe(true)
  expect(wrapper.find('[data-testid=\'lui-badge-text\']').text()).toBe(testText)
})

it('displays icon properly', () => {
  const testClass = 'ri-wifi-line'
  const wrapper = mount(LuiBadge, {
    slots: {
      icon: `<i class=${testClass}></i>`,
    },
  })
  expect(wrapper.find('[data-testid=\'lui-badge-icon\']').exists()).toBe(true)
  expect(
    wrapper.find('[data-testid=\'lui-badge-wrapper\']').find('i').classes(),
  ).toContain(testClass)
})

it('locates correctly based position prop', () => {
  const positions = [
    {
      position: 'top-left',
      expectedClasses: { vertical: 'top-0', horizontal: 'left-0' },
    },
    {
      position: 'bottom-left',
      expectedClasses: { vertical: 'bottom-0', horizontal: 'left-0' },
    },
    {
      position: 'top-right',
      expectedClasses: { vertical: 'top-0', horizontal: 'right-0' },
    },
    {
      position: 'bottom-right',
      expectedClasses: { vertical: 'bottom-0', horizontal: 'right-0' },
    },
  ]

  positions.forEach((positionOption) => {
    const combinedComponents = {
      template: `
          <lui-badge position=${positionOption.position} text="9">
            <lui-avatar color="secondary" :rounded="true">
              <template #icon>
                <i class="ri-user-fill"></i>
              </template>
            </lui-avatar>
          </lui-badge>

          `,
      components: {
        LuiAvatar,
        LuiBadge,
      },
    }
    const wrapper = mount(combinedComponents)
    // Getting the desired div for checking if it is located correctly
    const absoluteBadgeClasses = wrapper
      .find('[data-testid=\'lui-badge-wrapper\']')
      .classes()
    // Check necessary classes
    expect(absoluteBadgeClasses).toContain('absolute')
    expect(absoluteBadgeClasses).toContain(
      positionOption.expectedClasses.vertical,
    )
    expect(absoluteBadgeClasses).toContain(
      positionOption.expectedClasses.horizontal,
    )
  })
})
