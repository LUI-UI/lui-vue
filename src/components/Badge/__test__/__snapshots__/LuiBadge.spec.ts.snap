// Vitest Snapshot v1

exports[`Default LuiBadge > renders properly based on default values 1`] = `
"<div data-testid=\\"lui-badge\\" class=\\"lui-badge relative inline-block\\">
  <div data-testid=\\"lui-badge-wrapper\\" class=\\"flex rounded-full border border-solid justify-center items-center w-2.5 h-2.5 text-sm bg-primary-500 text-white border-transparent\\">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;
